docs/cvl/cvl2/changes.md

```diff
@@ -0,0 +1,583 @@
+for example `to_uint256(x)`.
+
+In CVL 2, there are now two casting operators: `assert_<type>(value)` and
+`require_<type>(value)` (for example: `assert_uint8(x)` or `require_uint8(x)`).
+Each of these cases checks that the value is in range; the `assert` cast will
+report a counterexample if the value is out of range, while the `require` cast
+will ignore counterexamples where the cast value is out of range.
+
+```{todo}
+is it an error to use an explicit cast for an upcast?
+```
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### Modulo operator `%` returns negative values for negative inputs
+
+As in Solidity, if `n < 0` then `n % k == -(-n % k)`.
+
+### Support for `bytes1`...`bytes32`
+
+CVL 2 supports the types `bytes1`, `bytes2`, ..., `bytes32`, as in Solidity.
+Number literals must be explicitly cast to these types using `to_bytesN`; for
+example:
+
+```cvl
+bytes32 x = to_bytes32(0);
+```
+
+There is no way to convert between these types and integer types (except for
+literals as just mentioned).
+
+```{todo}
+Update this if we add casting between `uint256` and `bytes32`.
+```
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+(cvl2-bitwise)=
+### Changes for bitwise operations
+
+In CVL1, the exact details for bitwise operations (such as `&`, `|`, and `<<`) were not
+completely specified, especially for negative integers.
+
+In CVL 2, all bitwise operations (`&`, `|`, `~`, `>>`, `>>>`, `<<`, and `xor`)
+first convert to `uint256`, then perform the operations on the full 256-bit
+word, then convert back to the expected type.  Signed integer types use
+twos-complement encoding.
+
+The two right-shifts differ in how they treat signed integers.  `>>` is an
+arithmetic shift; it preserves the sign bit.  `>>>` is a logical shift; it pads
+the shifted word with zero.
+
+Bitwise operations cannot be performed on `mathint` values.
+
+### Conversion between `bytes<k>`, `address`, and integer types
+
+```{todo}
+finish
+```
+
+Removed features
+----------------
+
+As part of the transition to CVL 2.0, we have removed several language features
+that are no longer used.
+
+We have removed these features because we think they are no longer used and no
+longer useful.  If you find that you do need one of these features, contact
+Certora support.
+
+### Methods entries for sighashes
+
+In CVL 1, you could write a sighash instead of a method identifier in the
+`methods` block.  This feature is no longer supported.  You will need to have
+the name and argument types of the called method in order to provide an entry.
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### `invoke`, `sinvoke`, and `call`
+
+Older versions of CVL had special syntax for calling contract functions:
+ - `invoke f(args);` should be replaced with `f(args);`.
+ - `sinvoke f(args);` should be replaced with `f@withrevert(args);`
+ - `call f(args)` should be replaced with `f(args)`.
```

Comment:
this was not for contract functions, it was for CVL Functions (formerly called "subroutines")